File: src/ObjectManager/Object.Tes/FilePacks/Records/040-HAIR.Hair.cs
using OA.Core;

namespace OA.Tes.FilePacks.Records
{
    public class HAIRRecord : Record
    {
        public override string ToString() => $"HAIR: {EDID.Value}";
        public STRVField EDID { get; set; } // Editor ID
        public STRVField FULL;
        public MODLGroup MODL;
        public FILEField ICON;
        public BYTEField DATA; // Playable, Not Male, Not Female, Fixed

        public override bool CreateField(UnityBinaryReader r, GameFormatId format, string type, int dataSize)
        {
            switch (type)
            {
                case "EDID": EDID = new STRVField(r, dataSize); return true;
                case "FULL": FULL = new STRVField(r, dataSize); return true;
                case "MODL": MODL = new MODLGroup(r, dataSize); return true;
                case "MODB": MODL.MODBField(r, dataSize); return true;
                case "ICON": ICON = new FILEField(r, dataSize); return true;
                case "DATA": DATA = new BYTEField(r, dataSize); return true;
                default: return false;
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/040-HAIR.Hair.swift
//
//  HAIRRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class HAIRRecord: Record, IHaveEDID, IHaveMODL {
    public override var description: String { return "HAIR: \(EDID)" }
    public var EDID: STRVField = STRVField_empty // Editor ID
    public var FULL: STRVField!
    public var MODL: MODLGroup? = nil
    public var ICON: FILEField!
    public var DATA: BYTEField! // Playable, Not Male, Not Female, Fixed
    
    override func createField(_ r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        switch type {
        case "EDID": EDID = r.readSTRV(dataSize)
        case "FULL": FULL = r.readSTRV(dataSize)
        case "MODL": MODL = MODLGroup(r, dataSize)
        case "MODB": MODL!.MODBField(r, dataSize)
        case "ICON": ICON = r.readSTRV(dataSize)
        case "DATA": DATA = r.readT(dataSize)
        default: return false
        }
        return true
    }
}
